be-java-quarkus/Jenkinsfile

```diff
@@ -0,0 +1,80 @@
+def odsNamespace = env.ODS_NAMESPACE ?: 'ods'
+def odsGitRef = env.ODS_GIT_REF ?: 'master'
+def odsImageTag = env.ODS_IMAGE_TAG ?: 'latest'
+
+library("ods-jenkins-shared-library@${odsGitRef}")
+
+odsQuickstarterPipeline(
+    imageStreamTag: "${odsNamespace}/jenkins-slave-maven:${odsImageTag}",
+    // maven-slave selects the java version based on available memory, to ensure java11 we raise the requested memory
+    resourceLimitMemory: "2Gi",
+    resourceRequestMemory: "1Gi",
+) { context ->
+
+  stage('Build Quarkus project') {
+
+    // TODO: remove this once maven-slave uses maven wrapper (probably released with mvn version 3.7)
+    // install maven
+    def mavenVersion = "3.6.3"
+    def mavenBaseUrl = "https://apache.osuosl.org/maven/maven-3/${mavenVersion}/binaries"
+    sh """
+      mkdir -p /tmp/maven /tmp/maven/ref
+      curl -fsSL -o /tmp/apache-maven.tar.gz ${mavenBaseUrl}/apache-maven-${mavenVersion}-bin.tar.gz
```

Comment:
why we don't use the maven that comes from the slave image? is not running on the version we wants? then maybe we should update the maven image?

Reply:
Quarkus requires a more recent version than we provide in the slave image. AFAIK in one of the next bigger maven updates they add sort of a wrapper that we already know from gradle, with that it would be easier to just rely on that, but since this not there yet I just added an additional maven version here.

Yes, alternatively we could update the maven in the slave image, but since I'm not quite sure about backwards compatibility of maven versions, I didn't want to open that box just because of a quickstarter in the extra-quickstarter repo. 
But I think it is not a bad idea anyway to update these things and I'll create a PR in ods-quickstarters for it. If this will happen I can still change it for the quarkus quickstarter afterwards.

Reply:
https://github.com/opendevstack/ods-quickstarters/issues/484

Reply:
ok switched jenkins agent from maven to graalvm, which had the latest maven version anyway. With that I was able to remove the entire maven part from the Jenkinsfile